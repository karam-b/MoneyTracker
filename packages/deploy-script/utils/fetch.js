/**
 * @typedef {(cb?: ((a:safeAny) => any)) => Promise<any>} Get
 */

/**
 * @type {(...args: Parameters<_fetch>) => Promise<Response & { get : Get }> }
 */
exports.fetch = async (...args) =>
  import('node-fetch')
    .then((m) => m.default)
    .then((fetch) => fetch(...args))
    .then(async (res) => {
      var body
      try {
        body = await res.json()
      } catch {}
      Object.defineProperty(res, 'get', {
        value: async function (acc) {
          if (!res.ok) {
            // ???
          }
          return typeof acc === 'function' ? acc(body) : body
        },
      })
      return res
    })

exports._fetch = (...args) =>
  import('node-fetch').then((m) => m.default(...args))

/**
 * possible undefined but can be optionally chained
 *
 * i.e. :give an error when you try to access property but works fine when you use optional chaining
 *
 * note: this is how optional chaining shoud work with `unknown` but idw and I don`t wanna bother
 * @example safeAny.hi.hi.hi.hi // error
 * @example safeAny?.hi?.hi?.hi?.hi // works
 * @example safeAny?.hi === "random" // works
 * @typedef {undefined | ({[k: string]: safeAny} & (boolean | string | number | function))} safeAny
 */
