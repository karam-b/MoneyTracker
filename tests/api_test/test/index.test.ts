import fetch from '../helpers/fetch'

it('root', async () => {
  const test = await fetch('/')

  expect(test).toMatchSnapshot()
})
