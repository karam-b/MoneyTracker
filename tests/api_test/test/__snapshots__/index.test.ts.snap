// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`root 1`] = `Promise {}`;
